client/src/components/activities/ActivityFilter/lib/filter-name.ts

```diff
@@ -0,0 +1,33 @@
+import type { ActivityFilterWithValues } from "@/components/activities/ActivityFilter/ActivityFilter.types";
+import type { ActivityWithIds } from "@t/data/activity.types";
+
+type ActivityFilterNameType = ActivityFilterWithValues["name"]["type"];
+
+export const namePredicates: Record<
+	ActivityFilterNameType,
+	(name: string, value: string) => boolean
+> = {
+	includes: (name, value) => name.includes(value),
+	equals: (name, value) => name === value,
+	excludes: (name, value) => !name.includes(value),
+	startsWith: (name, value) => name.startsWith(value),
+	endsWith: (name, value) => name.endsWith(value)
+};
+
+export const nameTypeOptions = Object.keys(namePredicates) as ActivityFilterNameType[];
+
+export function filterByName(
+	activities: ActivityWithIds[],
+	filter: ActivityFilterWithValues["name"]
+): ActivityWithIds[] {
+	const value = filter.value;
```

Comment:
with only 2 uses, I don't think this needs to be in an intermediate variable

Reply:
Need to use a variable, otherwise the predicate thinks the value may be nullable (because inside the filter function, it can't guarantee that it isn't).